src/wallet/wallet2.cpp

```diff
@@ -572,23 +574,63 @@ size_t estimate_rct_tx_size(int n_inputs, int mixin, int n_outputs, size_t extra
   // txnFee
   size += 4;
 
-  LOG_PRINT_L2("estimated rct tx size for " << n_inputs << " with ring size " << (mixin+1) << " and " << n_outputs << ": " << size << " (" << ((32 * n_inputs/*+1*/) + 2 * 32 * (mixin+1) * n_inputs + 32 * n_outputs) << " saved)");
+  LOG_PRINT_L2("estimated " << (bulletproof ? "bulletproof" : "borromean") << " rct tx size for " << n_inputs << " inputs with ring size " << (mixin+1) << " and " << n_outputs << " outputs: " << size << " (" << ((32 * n_inputs/*+1*/) + 2 * 32 * (mixin+1) * n_inputs + 32 * n_outputs) << " saved)");
   return size;
 }
 
 size_t estimate_tx_size(bool use_rct, int n_inputs, int mixin, int n_outputs, size_t extra_size, bool bulletproof)
 {
   if (use_rct)
-    return estimate_rct_tx_size(n_inputs, mixin, n_outputs + 1, extra_size, bulletproof);
+    return estimate_rct_tx_size(n_inputs, mixin, n_outputs, extra_size, bulletproof);
   else
     return n_inputs * (mixin+1) * APPROXIMATE_INPUT_BYTES + extra_size;
 }
 
+uint64_t estimate_tx_weight(bool use_rct, int n_inputs, int mixin, int n_outputs, size_t extra_size, bool bulletproof)
+{
+  size_t size = estimate_tx_size(use_rct, n_inputs, mixin, n_outputs, extra_size, bulletproof);
```

Comment:
I think the BP part in the `estimate_tx_size`:
```c++
   if (bulletproof)
     size += ((2*6 + 4 + 5)*32 + 3) * n_outputs;
```
is not correct.

First of all, this is supposed to be an estimate of the base size with 2 outputs, right?

I don't understand the meaning of `+ 3` in the formula.

And for 2 outputs aggregated into one BP, the formula should instead be:
```c++
   if (bulletproof)
     size += (2*7 + 4 + 5)*32;
```

Reply:
It\s meant to be an etsimate of the size for n_output outputs. It needs to be modified for multi output proofs. The +3 is overhead for array sizes.

Reply:
Ok